src/arrayMethodSort.js

```diff
@@ -5,8 +5,63 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let currentValue;
+    let previousValue;
+    let isSorted;
+
+    if (compareFunction !== undefined) {
+      while (!isSorted) {
+        isSorted = true;
+
+        for (let i = 1; i < this.length; i++) {
+          previousValue = this[i - 1];
+          currentValue = this[i];
+
+          if (compareFunction(previousValue, currentValue) > 0) {
+            this[i] = previousValue;
+            this[i - 1] = currentValue;
+            isSorted = false;
+          }
+        }
+      }
+
+      return this;
+    }
```

Comment:
DRY, you do the same operations in both cases, the only difference is a compare function. Create a default compare function and assign it as a default parameter if a function isn't passed to sort